AutomatedTesting/Gem/PythonTests/Atom/tests/hydra_AtomEditorComponents_DepthOfFieldAdded.py

```diff
@@ -175,29 +326,404 @@ def AtomEditorComponents_DepthOfField_AddedToEntity():
+
+        # 24. EnabledDebugColoring Override can be turned on and off.
+        depth_of_field_component.set_component_property_value(
+            AtomComponentProperties.depth_of_field('EnableDebugColoring Override'), True)
+        general.idle_wait_frames(1)
+        Report.result(
+            Tests.enabledebugcoloring_override_on,
+            depth_of_field_component.get_component_property_value(
+                AtomComponentProperties.depth_of_field('EnableDebugColoring Override')) is True)
+
+        depth_of_field_component.set_component_property_value(
+            AtomComponentProperties.depth_of_field('EnableDebugColoring Override'), False)
+        general.idle_wait_frames(1)
+        Report.result(
+            Tests.enabledebugcoloring_override_off,
+            depth_of_field_component.get_component_property_value(
+                AtomComponentProperties.depth_of_field('EnableDebugColoring Override')) is False)
+
+        # 25. Enable Depth of Field an be turned on and off.
+        depth_of_field_component.set_component_property_value(
+            AtomComponentProperties.depth_of_field('Enable Depth of Field'), True)
+        general.idle_wait_frames(1)
+        Report.result(
+            Tests.enable_depth_of_field_on,
+            depth_of_field_component.get_component_property_value(
+                AtomComponentProperties.depth_of_field('Enable Depth of Field')) is True)
+
+        depth_of_field_component.set_component_property_value(
+            AtomComponentProperties.depth_of_field('Enable Depth of Field'), False)
+        general.idle_wait_frames(1)
+        Report.result(
+            Tests.enable_depth_of_field_off,
+            depth_of_field_component.get_component_property_value(
+                AtomComponentProperties.depth_of_field('Enable Depth of Field')) is False)
+
+        # Enable Depth of Field again for subsequent tests.
+        depth_of_field_component.set_component_property_value(
+            AtomComponentProperties.depth_of_field('Enable Depth of Field'), True)
+        general.idle_wait_frames(1)
+        Report.result(
+            Tests.enable_depth_of_field_on,
+            depth_of_field_component.get_component_property_value(
+                AtomComponentProperties.depth_of_field('Enable Depth of Field')) is True)
+
+        # 26. Quality Level can be set to 0 and back to 1.
+        depth_of_field_component.set_component_property_value(
+            AtomComponentProperties.depth_of_field('Quality Level'), 0)
+        general.idle_wait_frames(1)
+        Report.result(
+            Tests.quality_level_set_to_0,
+            depth_of_field_component.get_component_property_value(
+                AtomComponentProperties.depth_of_field('Quality Level')) == 0)
+
+        depth_of_field_component.set_component_property_value(
+            AtomComponentProperties.depth_of_field('Quality Level'), 1)
+        general.idle_wait_frames(1)
```

Comment:
you can remove more than half the general.idle_wait_frames(1) in this script. I didn't count them, but your probably waiting a total of a couple seconds. if you remove them from the second set on each property you should be fine. you mostly need these for the engine to render changes you apply or to have time for the engine to react to changes you apply. in most cases you are just reading that the value was set which doesn't require frames of game loop.

Reply:
Deleting half caused my script to hit an exception at the end. After some trial and error I was able to delete about 40%, not sure if this was worth the hassle. but it is fixed